awx/main/models/unified_jobs.py

```diff
@@ -1421,27 +1405,59 @@ def _build_job_explanation(self):
             return 'Previous Task Canceled: {"job_type": "%s", "job_name": "%s", "job_id": "%s"}' % (self.model_to_str(), self.name, self.id)
         return None
 
+    def fallback_cancel(self):
+        if not self.celery_task_id:
+            self.refresh_from_db(fields=['celery_task_id'])
+        self.cancel_dispatcher_process()
+
+    def cancel_dispatcher_process(self):
+        """Returns True if dispatcher running this job acknowledged request and sent SIGTERM"""
+        if not self.celery_task_id:
+            return
+        canceled = []
+        try:
+            # Use control and reply mechanism to cancel and obtain confirmation
+            timeout = 5
+            canceled = ControlDispatcher('dispatcher', self.controller_node).cancel([self.celery_task_id])
+        except (socket.timeout, RuntimeError):
```

Comment:
It's pretty dangerous to include both of these exception types here. We should catch `RuntimeError` separately and use `logger.exception`. Right now this is eating:

```
tools_awx_1     | 2022-08-25 16:12:37,974 ERROR    [bf11a9d4] awx.main.models.unified_jobs could not reach dispatcher on awx_1 within 5s
tools_awx_1     | Traceback (most recent call last):
tools_awx_1     |   File "/awx_devel/./awx/main/models/unified_jobs.py", line 1420, in cancel_dispatcher_process
tools_awx_1     |     canceled = ControlDispatcher('dispatcher', self.controller_node).cancel([self.celery_task_id])
tools_awx_1     |   File "/awx_devel/./awx/main/dispatch/control.py", line 41, in cancel
tools_awx_1     |     return self.control_with_reply('cancel', *args, extra_data={'task_ids': task_ids}, **kwargs)
tools_awx_1     |   File "/awx_devel/./awx/main/dispatch/control.py", line 59, in control_with_reply
tools_awx_1     |     for reply in conn.events(select_timeout=timeout, yield_timeouts=True):
tools_awx_1     |   File "/awx_devel/./awx/main/dispatch/__init__.py", line 35, in events
tools_awx_1     |     raise RuntimeError('Listening for events can only be done in autocommit mode')
tools_awx_1     | RuntimeError: Listening for events can only be done in autocommit mode
```

Reply:
Never mind. This is also happening on devel. Yikes.

Reply:
https://github.com/ansible/awx/issues/12740